Put detail back to SCAN page. (#2535)
redis/redis-doc: commands/scan.md

However once the data structures are bigger and are promoted to use real hash tables, the `SCAN` family of commands will resort to the normal behavior. Note that since this special behavior of returning all the elements is true only for small aggregates, it has no effects on the command complexity or latency. However the exact limits to get converted into real hash tables are [user configurable](/topics/memory-optimization), so the maximum number of elements you can see returned in a single call depends on how big an aggregate data type could be and still use the packed representation.

Also note that this behavior is specific of `SSCAN`, `HSCAN` and `ZSCAN`. `SCAN` itself never shows this behavior because the key space is always represented by hash tables.

## Further reading

> For more information on `SCAN` please refer to the [The Redis Keyspace](/docs/manual/keyspace) tutorial.

Reviewer: So if we delete most of the same text from the keyspace tutorial (which was move there from scan command docs here: https://github.com/redis/redis-doc/commit/882c184514f88f048df3d773e300a39882773e9d#diff-6875ae7c6d2de818737cec8e3e3c0221294ab337dab94fbe7c05ad66e1f789ce) there is not much more information about scan there. Maybe instead something like this:

```suggestion
For more information about managing keys, please refer to the [The Redis Keyspace](/docs/manual/keyspace) tutorial.
```

And to avoid having duplicate information on two places, we can delete Scan basic usage, scan guarantees, etc. from this tutorial and instead link to the SCAN command from there.
